src/Common/src/Interop/OSX/System.Native/Interop.ProcRUsage.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        // Constants from sys/resource.h
+        private const int RUSAGE_SELF = 0;
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal unsafe struct RUsageInfo
```

Comment:
Are we assuming that these have a fixed layout because they're only used on OS X? Does FreeBSD have these structs, functions?